real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/nodes/localizers/LocalizerUtilTest.java

```diff
@@ -0,0 +1,70 @@
+package com.roboclub.robobuggy.nodes.localizers;
+
+import com.roboclub.robobuggy.ui.LocTuple;
+import org.junit.Test;
+
+import java.util.Map;
+
+import static java.lang.Math.PI;
+import static java.lang.Math.sqrt;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests the Localizer util functions
+ */
+public class LocalizerUtilTest {
+
+    private final double latPittsburgh = 40.440310;
+    private final double lonPittsburgh = -79.9471537;
+    private final double calcedDistOneDegMeters = 84721.0;
+
+    // the distance in meters for 1˚ up and 1˚ right
+    private final double diagDistMeters = 139505.0;
+
+    private static final double FEET_TO_METERS = 0.3048;
+
+    @Test
+    public void test_convertLatLngDeltaToMeters() throws Exception {
+
+        LocTuple pitt = new LocTuple(latPittsburgh, lonPittsburgh);
+        LocTuple oneLonRight = new LocTuple(latPittsburgh, lonPittsburgh + 1);
+        LocTuple diagonalRightAndUp = new LocTuple(latPittsburgh + 1, lonPittsburgh + 1);
+
+        // one to the right
+        Map.Entry<Double, Double> deltaAndHeading = LocalizerUtil.convertLatLngDeltaToMeters(pitt, oneLonRight);
+
+        assertEquals(deltaAndHeading.getKey(), calcedDistOneDegMeters, 500);
+        assertEquals(deltaAndHeading.getValue(), 0.0, 0.0000001);
+
+        // one right and one up
+        deltaAndHeading = LocalizerUtil.convertLatLngDeltaToMeters(pitt, diagonalRightAndUp);
+
+        assertEquals(diagDistMeters, deltaAndHeading.getKey(), 500);
+        assertEquals(PI/4, deltaAndHeading.getValue(), 0.01);
+
+    }
+
+    @Test
+    public void test_convertMetersToLatLng() throws Exception {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method names should not contain underscores](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2337031408/issues/source?bid=2938665&fileBranchId=3199623#l48)